lib/noether/maybe.ex

```diff
@@ -85,12 +89,6 @@ defmodule Noether.Maybe do
   """
   @spec choose(any(), fun1(), fun1()) :: any()
   def choose(a, f, g) when is_function(f, 1) and is_function(g, 1) do
-    b = f.(a)
-
-    if is_nil(b) do
-      g.(a)
-    else
-      b
-    end
+    maybe(a, f, map(a, g))
```

Comment:
I don't know how this works in elixir, but are you sure that `map(a, g)` is not evaluated before the call to `maybe`? If that's the case and someone is doing side effects on `g` we are doing an involuntary breaking change... Maybe we can write a test where `g` (improperly) throws an exception and `f` returns a non nil value to check for the correct behavior

Reply:
yes, I believe `map` will be evaluated before `maybe` since Elixir is not lazy. Very good point!

Reply:
reverted